src/lightning_app/core/api.py

```diff
@@ -114,6 +123,13 @@ def run_once(self):
         except queue.Empty:
             pass
 
+        if self.app_status_queue is not None:
```

Comment:
I don't think this is required.
We are already pulling the state above. Let's simply perform status comparaison there operation there.